### PR TITLE
Data marshalling (switch from protobuf to rapidjson)

### PR DESCRIPTION
**Pull request by robertasn**
## Description
Telemetry module is now fully included and uses RapidJSON instead of protobuf

 *Click up [link](https://app.clickup.com/t/2ekmgj)* 

## Changes
- Telemetry module is now fully included
- Telemetry module could be disabled with flag `--telemetry_off`
- Protobuf dependencies are removed
- RapidJSON is now used as data marshalling solution

## Additional Info 
N/A

### CODE REVIEW

**Review by mkristien — COMMENTED** [review]
I like the way rapidjson is integrated into the compilation framework. For me this is "approve" but somebody should still review the telemetry code.

File: src/telemetry/recvloop.cpp [hunk 1]
```diff
@@ -38,63 +38,41 @@ RecvLoop::RecvLoop(Logger &log, data::Data& data, Main* main_pointer)
...
+
   data::Telemetry telem_data_struct = data_.getTelemetryData();
 
   while (true) {
-    try {
-      msg = main_ref_.client_.receiveData();
-    }
-    catch (std::exception& e) {
-      log_.ERR("Telemetry", "%s", e.what());
+    std::string message = main_ref_.client_.receiveData();
```
Comment (neilweidinger) [inline src/telemetry/recvloop.cpp]: `receiveData()` can still throw an exception, probably best to put the try catch block back in here

File: src/telemetry/sendloop.cpp [hunk 2]
```diff
@@ -36,144 +39,246 @@ void SendLoop::run()
...
+    packEmergencyBrakesMessage(writer);
+    writer.EndObject();
+
+    if (!main_ref_.client_.sendData(sb.GetString())) {
+      log_.ERR("Telemetry", "Error sending message");
       data::Telemetry telem_data_struct = data_.getTelemetryData();
       telem_data_struct.module_status = data::ModuleStatus::kCriticalFailure;
       data_.setTelemetryData(telem_data_struct);
-
-      break;
```
Comment (neilweidinger) [inline src/telemetry/sendloop.cpp:60]: Why was the `break` removed here?

File: src/telemetry/sendloop.cpp [hunk 3]
```diff
@@ -36,144 +39,246 @@ void SendLoop::run()
...
+  writer.StartArray();
   for (bool brake_retracted : emergency_brakes_data.brakes_retracted) {
-    emergency_brakes_msg->add_brakes(brake_retracted);
+    writer.Bool(brake_retracted);
+  }
+  writer.EndArray();
+  writer.EndObject();
+}
+
+std::string SendLoop::convertStateMachineState(data::State state)
```
Comment (neilweidinger) [inline src/telemetry/sendloop.cpp:224]: We could put this in a `utils` file maybe later down the road, but not too important

**Review by neilweidinger — CHANGES_REQUESTED** [review]

File: src/telemetry/recvloop.cpp (hunk 1 above)
Comment (robertasn) [reply src/telemetry/recvloop.cpp]: Good point, I'll add that.

File: src/telemetry/sendloop.cpp (hunk 2 above)
Comment (robertasn) [reply src/telemetry/sendloop.cpp:60]: That's a silly copy-paste mistake. Thanks for spotting that!

File: src/telemetry/sendloop.cpp (hunk 3 above)
Comment (robertasn) [reply src/telemetry/sendloop.cpp:224]: Certainly. Also probably going to add some abstraction to the whole data packing, so you could specify the key, data type and value (together with max and min values) or array in one line.

**Comment by freddiejbawden** [comment]
Adding a small commit to bypass the coverage CI

**Review by neilweidinger — APPROVED** [review]

**Pull request merged**